packages/web-app-files/tests/unit/components/SideBar/Links/FileLinks.spec.js

```diff
@@ -0,0 +1,240 @@
+    })
+  }
+
+  function getShallowWrapper(store) {
+    return shallowMount(FileLinks, {
+      localVue,
+      store: store,
+      stubs: stubs,
+      provide: {
+        changeView: jest.fn()
+      }
+    })
+  }
+
+  function getMountedWrapper(store) {
+    return mount(FileLinks, {
+      localVue,
+      store: store,
+      stubs: {
+        'link-edit': true,
+        'list-item': true
+      },
+      provide: {
+        changeView: jest.fn()
+      }
+    })
+  }
+
+  describe('current view', () => {
+    describe('when view is to show', () => {
+      describe('links', () => {
+        describe('when links list is not empty', () => {
+          const store = createStore()
+          const wrapper = getShallowWrapper(store)
+
+          it('should render list of items of provided links', () => {
+            const linkListItems = wrapper.findAll(selectors.listItemStub)
+            expect(linkListItems.length).toBe(2)
+            expect(linkListItems.at(0).props().link).toMatchObject({
+              id: '1',
+              indirect: false,
+              name: 'public link 1',
+              key: 'direct-link-1'
+            })
+            expect(linkListItems.at(1).props().link).toMatchObject({
+              id: '2',
+              indirect: true,
+              name: 'public link 2',
+              key: 'direct-link-2'
+            })
+            expect(wrapper.find(selectors.linkNoResults).exists()).toBeFalsy()
+          })
+
+          it('should not show no results', () => {
+            expect(wrapper.find(selectors.linkNoResults).exists()).toBeFalsy()
+          })
+        })
+
+        it('should show no results if no links is provided', () => {
+          const wrapper = getShallowWrapper(createStore({ links: [] }))
+          expect(wrapper.find(selectors.linkNoResults).exists()).toBeTruthy()
+        })
+      })
+      describe('when linksLoading is true', () => {
+        it('should show oc loader', () => {
+          const store = createStore({ currentFileOutgoingSharesLoading: true })
+          const wrapper = getShallowWrapper(store)
+
+          expect(wrapper.find('oc-loader-stub').exists()).toBeTruthy()
+          expect(wrapper.find('oc-loader-stub').attributes().arialabel).toBe(
+            'Loading list of file links'
+          )
+        })
+      })
+      describe('when linksLoading is false', () => {
+        it('should not show oc loader', () => {
+          const store = createStore()
+          const wrapper = getShallowWrapper(store)
+
+          expect(wrapper.find('oc-loader-stub').exists()).toBeFalsy()
+        })
+
+        describe('when canCreatePublicLinks is true', () => {
+          it('should show link add button', () => {
+            const store = createStore()
+            const wrapper = getShallowWrapper(store)
+
+            expect(wrapper.find(selectors.linkAddButton).exists()).toBeTruthy()
+          })
+
+          describe('when add new link button is clicked', () => {
```

Comment:
```suggestion
          describe('when add-new-link button is clicked', () => {
```

Reply:
```suggestion
          describe('when the add-new-link button is clicked', () => {
```

And put "the". (I probably need to look at all the `describe` text if I am going to be fussy about the English grammar!)